docs: integrate webserver configuration

PowerKiKi opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
The article previously referred was out of date. Most of the article content
was already duplicated here, or described how to generate multiple bundles. But
creating multiple bundles manually is now obsolete and the parameter `--localize`
should be preferred instead, because it is much faster.


Issue Number: N/A


## What is the new behavior?


This commit integrate the only interesting part left, relating to webserver configuration.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

PowerKiKi committed:
docs: integrate webserver configuration

The article previously referred was out of date. Most of the article content
was already duplicated here, or described how to generate multiple bundles. But
creating multiple bundles manually is now obsolete and the parameter `--localize`
should be preferred instead, because it is much faster.

This commit integrate the only interesting part left, relating to webserver configuration.

Co-authored-by: David Shevitz <dshevitz@google.com>

IgorMinar reviewed (approved):
LGTM! thank you

IgorMinar commented:
@PowerKiKi can you please rebase this PR?

petebacondarwin commented on aio/content/guide/i18n.md:
```diff
@@ -777,4 +777,71 @@ You specify the warning level in the `options` section for the `build` target of
 ### Deployment for multiple locales
 
-For more details about how to create scripts to generate an app in multiple languages and how to set up Apache 2 and NGINX to serve them from different subdirectories, read [this tutorial by Philippe Martin](https://dev.to/angular/deploying-an-i18n-angular-app-with-angular-cli-2fb9).
+Typical deployment of multiple languages will serve each language from a different subdirectory. The user will be redirected to the preferred language defined in the browser, or `fr` if a preferred language is not found. And the user will still be able to change language by navigating to other subdirectories.
```
> or `fr` if a preferred language is not found

This implies that `fr` is always the default. Can you re-word this to indicate that this is not always the case?

Also I note that this is one very long line! Can you add some line-breaks to keep the lines down to around 100 chars in length?

PowerKiKi replied:
resolved in rebase

petebacondarwin commented on aio/content/guide/i18n.md:
````diff
@@ -777,4 +777,71 @@ You specify the warning level in the `options` section for the `build` target of
+```
+http {
+
+    # Browser preferred language detection (does NOT requires AcceptLanguageModule)
````
```suggestion
    # Browser preferred language detection (does NOT require AcceptLanguageModule)
```

PowerKiKi commented:
I rebased and fixed the things mentioned.

I am not sure where this new "Configuring servers" section should live. In `i18n.md` because it is only relevant to i18n ? or in `deployment.md` because it is related to deployment (but deployment is already mentioned in `i18n.md` anyway) ?

petebacondarwin reviewed (approved):
Thanks for the fixups @PowerKiKi - I agree I am not confident with where this should go. Let's ask @aikidave to chime in.

aikidave commented:
My initial assessment is to put this content in i18n.md, as that's where it's most relevant.

However, it might be useful to add a link from deployment.md. I don't think that's necessary for this PR, however.

AndrewKushnir commented:
Hi @aikidave, just want to double check if we can proceed with merging this PR? Thank you.

aikidave commented on aio/content/guide/i18n.md:
```diff
@@ -774,8 +774,82 @@ The HTML `base` tag with the `href` attribute specifies the base URI, or URL, fo
-For details on how to deploy apps to a remote server, see [Deployment](guide/deployment "Deployment guide").
+### Configuring servers
+
+Typical deployment of multiple languages will serve each language from a different subdirectory.
+The user will be redirected to the preferred language defined in the browser (via the
+`Accept-Language` HTTP header), or it will fallback to a default language if a preferred language
+is not found. And the user will still be able to change language by navigating to other subdirectories,
+most likely via a some kind of menu implemented in the application.
```
```suggestion
Typical deployment of multiple languages serve each language from a different subdirectory.
Users are redirected to the preferred language defined in the browser using the
`Accept-Language` HTTP header. If the user has not defined a preferred language, the browser uses the default language. Users can change the language by navigating to other subdirectories, which often occurs using a menu implemented in the application.
```

In general, avoid using "will" in sentences. Instead, use present tense.

Also, words such as "it" or "this" on their own are ambiguous. For example, "...it will fallback. " What does "it" refer to? I made a guess here--let me know if it's not accurate.

Otherwise, I made a few changes to make the text more concise.

aikidave commented on aio/content/guide/i18n.md:
```diff
@@ -774,8 +774,82 @@ The HTML `base` tag with the `href` attribute specifies the base URI, or URL, fo
+is not found. And the user will still be able to change language by navigating to other subdirectories,
+most likely via a some kind of menu implemented in the application.
+
+For other general advice on how to deploy apps to a remote server, see [Deployment](guide/deployment "Deployment guide").
```
```suggestion
For more information on how to deploy apps to a remote server, see [Deployment](guide/deployment "Deployment guide").
```

Updated to use standard phrasing.

aikidave commented on aio/content/guide/i18n.md:
```diff
@@ -774,8 +774,82 @@ The HTML `base` tag with the `href` attribute specifies the base URI, or URL, fo
+
+#### Nginx
+
+The Nginx configuration could look something like the following:
```
```suggestion
The following is an example of an Nginx configuration.
```

aikidave commented on aio/content/guide/i18n.md:
```diff
@@ -774,8 +774,82 @@ The HTML `base` tag with the `href` attribute specifies the base URI, or URL, fo
+
+#### Apache
+
+The Apache configuration could look something like the following:
```
```suggestion
The following is an example of an Apache configuration.
```

aikithoughts reviewed (approved):
Reviewed-for: global-docs-approvers

Some change to consider to improve readability and clarity. Otherwise, this seems fine!

AndrewKushnir commented:
Thanks @aikidave! I've added the "cleanup" label for now and will remove it once your comments are addressed. Thank you.

AndrewKushnir commented:
@PowerKiKi thanks for updating the PR. @aikidave could you please have another look?

jessicajaniuk commented:
@PowerKiKi It looks like this PR needs a rebase. The CircleCI job is failing. You can find additional information and instructions [here](https://app.circleci.com/pipelines/github/angular/angular/25151/workflows/a25bf112-04c0-4e70-b513-6e3d3aac7c07/jobs/866186). Thank you.

aikithoughts reviewed (approved):
Looks great! Thank you!

Reviewed-for: global-docs-approvers
